map_nav_node:
  ros__parameters:
    wms:
      url: 'http://localhost:8080/wms'
      version: '1.1.1'
      layer: 'Imagery'
      srs: 'EPSG:4326'  # don't change this setting, internal logic may often implicitly assume EPSG:4326
      request_timeout: 5
    misc:
      mock_gps_selection: 1
      export_position: 'position.json'
      export_projection: 'projection.json'
      max_pitch: 30  # Used by _should_match
      variance_estimation_length: 20
      min_match_altitude: 80
      blur_threshold: 1.5  # standard deviations
      blur_window_length: 100
      min_matches: 15
    map_update:
      initial_guess: # Values from ksql_airport.world file
        - 37.523640
        - -122.255122
      update_delay: 1  # Keep low (e.g. 1 second) if vehicle flying fast or gimbal_projection: True
      default_altitude: 130
      gimbal_projection: True
      max_pitch: 30  # Applies if gimbal_projection: True, used by _should_update_map
      max_map_radius: 1000
      map_radius_meters_default: 400
      update_map_center_threshold: 50
      update_map_radius_threshold: 50
    matcher:
      #class: 'python_px4_ros2_map_nav.keypoint_matchers.superglue.SuperGlue'  # Imported dynamically
      #params_file: 'config/superglue_params.yml' # For parsing args for matcher's static initializer method
      class: 'python_px4_ros2_map_nav.keypoint_matchers.loftr.LoFTRMatcher'
      params_file: 'config/loftr_params.yml'
